core/src/main/java/com/microsoft/applicationinsights/internal/persistence/PersistenceHelper.java

```diff
@@ -0,0 +1,89 @@
+package com.microsoft.applicationinsights.internal.persistence;
+
+import com.microsoft.applicationinsights.internal.util.LocalFileSystemUtils;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.UUID;
+
+final class PersistenceHelper {
+
+    private static final Logger logger = LoggerFactory.getLogger(PersistenceHelper.class);
+
+    private static final long MAX_FILE_SIZE_IN_BYTES = 52428800; // 50MB
+    private static final String PREFIX = "transmission";
+    static final String PERMANENT_FILE_EXTENSION = ".trn";
+    static final String TEMPORARY_FILE_EXTENSION = ".tmp";
+
+    /**
+     * Windows: C:\Users\{USER_NAME}\AppData\Local\Temp\applicationinsights
+     * Linux: /var/temp/applicationinsights
+     */
+    static final File DEFAULT_FOlDER = new File(LocalFileSystemUtils.getTempDir(), "applicationinsights");
+
+    static File createTempFileWithUniqueName() {
+        File file = null;
+        try {
+            String prefix = PREFIX + "-" + System.currentTimeMillis() + "-" + UUID.randomUUID().toString().replaceAll("-", "");
+            file = File.createTempFile(prefix, null, DEFAULT_FOlDER);
```

Comment:
I don't think you need UUID here, because `createTempFile` will guarantee uniqueness for you